src/main/java/Item.java

```diff
@@ -0,0 +1,53 @@
+public abstract class Item {
+    private final String name; // is this final??? // the name of the Item
+    private int num; // the number of items in Item
+
+    /**
+     * Creates a new Item object.
+     * This constructor takes in a single String as and a single int as arguments..
+     * @param name the name of the Item.
+     * @param num the number of items in the Item object.
+     */
+    public Item(String name, int num) {
+        this.name = name;
+        this.num = num;
+    }
+
+    public Item(String name){
+        this.name = name;
+    }
+
+    /**
+     * Returns the name of the Item.
+     * @return the name of the Item.
+     */
+    public String getName(){return this.name;}
+
+    /**
+     * Returns the number of items in the Item object.
+     * @return the number of items in the Item object.
+     */
+    public int getNum(){return this.num;}
+
+    /**
+     * Decreases or increases num.
+     * @param be_added a negative or positive integer to add to num.
+     */
+    public void changeNum(int be_added) {
+        this.num = this.num + be_added;
+    }
+
+
+    public boolean checkMinusAvailable(int be_minus){
+        return this.num - be_minus >= 0;
+    }
```

Comment:
I think you could make checkMinusAvailable to private, and implement it in changeNum because I'm guessing, you are trying to set things to 0 if changing the amount makes it negative. So I don't think anyone else would use this method.